lib/Injections/NetworkAgent.js

```diff
@@ -0,0 +1,300 @@
+    if (typeof response.headers['content-type'] != 'string')
+      return 'text/plain';
+
+    return response.headers['content-type'].split(';')[0];
+  }
+
+  function constructRequestInfo(request) {
+    var requestId = request.__inspector_ID__;
+    var documentURL = request.__inspector_documentURL__;
+    var timestamp = request.__inspector_timing__.json.requestTime;
+
+    return {
+      requestId: requestId,
+      loaderId: process.pid + '',
+      documentURL: documentURL,
+      type: 'XHR',
+      request: {
+        headers: renderHeaders(request),
+        method: request.method,
+        postData: '',
+        url: documentURL
+      },
+      timestamp: timestamp,
+      initiator: {
+        stackTrace: getStackTrace(),
+        type: 'script'
+      }
+    };
+  }
+
+  function constructResponseInfo(response) {
+    var request = response.req;
+
+    return {
+      requestId: request.__inspector_ID__,
+      loaderId: process.pid + '',
+      timestamp: timestamp(),
+      type: 'XHR',
+      response: {
+        url: request.__inspector_documentURL__,
+        status: response.statusCode,
+        statusText: response.statusMessage,
+        headers: getCorrectHeaders(response),
+        mimeType: getMimeType(response),
+        connectionReused: request.shouldKeepAlive,
+        connectionId: request.connection.__inspector_ID__,
+        encodedDataLength: -1,
+        fromDiskCache: false,
+        fromServiceWorker: false,
+        timing: request.__inspector_timing__.json,
+        headersText: restoreHeadersText(response),
+        requestHeaders: renderHeaders(request),
+        requestHeadersText: request._header
+      }
+    };
+  }
+
+  debug.registerEvent('Network.requestWillBeSent');
+  debug.registerEvent('Network.responseReceived');
+  debug.registerEvent('Network._responseReceived');
+  debug.registerEvent('Network.dataReceived');
+  debug.registerEvent('Network._dataReceived');
+  debug.registerEvent('Network.loadingFinished');
+  debug.registerEvent('Network._loadingFinished');
+
+  wrapHttpRequests();
+  debug.once('close', unwrapHttpRequests);
+
+  function wrapHttpRequests() {
+    var lastRequestId = 0,
+        lastConnectionId = 0;
+
+    function handleHttpRequest() {
+      var timing = new Timing();
+
+      this.__inspector_timing__ = timing;
+      this.__inspector_ID__ = '' + lastRequestId++;
+      this.__inspector_documentURL__ = this._headers.host + this.path;
+
+      this.once('socket', handleSocket);
+      this.once('response', handleHttpResponse);
+      handleRequestEnd(this);
+
+      deferToConnect.apply(this, arguments);
+      timing.dnsStart();
+    }
+
+    function handleRequestEnd(request) {
+      var oldWrite = request.write,
+          oldEnd = request.end;
+
+      var requestInfo = constructRequestInfo(request);
+
+      request.write = function(chunk) {
+        oldWrite.apply(this, arguments);
+        requestInfo.request.postData += chunk || '';
```

Comment:
It might be a good idea to at least call `(chunk || '').toString('utf8')` to support buffers*. I'm not sure how to handle non-text though (e.g. when uploading an image in the body, not using multi-part).

(*) Or to fully copy concat-stream's implementation of collecting an array with all chunks and then to use `Buffer.concat` or `[].concat`.